libgambatte/include/gambatte-c.h

```diff
@@ -0,0 +1,160 @@
+//
+//   Copyright (C) 2023 by Pierre Wendling <pierre.wendling.4@gmail.com>
+//
+//   This program is free software; you can redistribute it and/or modify
+//   it under the terms of the GNU General Public License version 2 as
+//   published by the Free Software Foundation.
+//
+//   This program is distributed in the hope that it will be useful,
+//   but WITHOUT ANY WARRANTY; without even the implied warranty of
+//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//   GNU General Public License version 2 for more details.
+//
+//   You should have received a copy of the GNU General Public License
+//   version 2 along with this program; if not, write to the
+//   Free Software Foundation, Inc.,
+//   51 Franklin St, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+
+/* C interface for using a shared libgambatte */
+
+#ifndef GAMBATTE_C_H
+#define GAMBATTE_C_H
+
+#include "config.h"
+
+#include "gambatte.h"
+#include "newstate.h"
+
+#if defined(_WIN32) || defined(__CYGWIN__)
+#define GBEXPORT __declspec(dllexport)
```

Comment:
If this is used by the user of a dll rather than the dll itself, then dllexport would be incorrect, dllimport is the correct declspec (and I don't think visibility attribrute is correct either for non-Windows cases, rather it should just be `extern`

Reply:
Checking GCC's documentation on [visibility](https://gcc.gnu.org/wiki/Visibility), you're correct about the dllexport/dllimport.

According to that same documentation, for the non-Windows case, `__attribute__((visibility("default")))` seems to indicate both export and import.

Reply:
Seems like the change in https://github.com/pokemon-speedrunning/gambatte-core/pull/19/commits/d1ae9ec4a2bb31f66ae7b75f2287b607aa8a7004 [broke CI](https://github.com/FtZPetruska/gambatte-core/actions/runs/5167158457/jobs/9307863094). 

Looking around, other projects such as SDL seem to only be using [dllexport](https://github.com/libsdl-org/SDL/blob/SDL2/include/begin_code.h#L56).

Reply:
Breaking CI likely indicates it got used internally in the library, or it got used when linking it in statically. You can only use dllimport when you are a user of the library (so not the library itself) and the library is a dll (if it's static, you don't use dllimport/dllexport).